lib/web/scripts/desktop/configure-ad.ps1

```diff
@@ -0,0 +1,170 @@
+Unicode=yes
+[Version]
+signature="$CHICAGO$"
+[Service General Setting]
+"SCardSvr",2,""
+'@
+
+$COMMENT_XML=@'
+<?xml version='1.0' encoding='utf-8'?>
+<policyComments xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" revision="1.0" schemaVersion="1.0" xmlns="http://www.microsoft.com/GroupPolicy/CommentDefinitions">
+  <policyNamespaces>
+    <using prefix="ns0" namespace="Microsoft.Policies.TerminalServer"></using>
+  </policyNamespaces>
+  <comments>
+    <admTemplate></admTemplate>
+  </comments>
+  <resources minRequiredRevision="1.0">
+    <stringTable></stringTable>
+  </resources>
+</policyComments>
+'@
+
+
+$ACCESS_POLICY_GUID=((Get-GPO -Name $ACCESS_GPO_NAME).Id.Guid).ToUpper()
+$ACCESS_GPO_PATH="$env:SystemRoot\SYSVOL\sysvol\$DOMAIN_NAME\Policies\{$ACCESS_POLICY_GUID}\Machine\Microsoft\Windows NT\SecEdit"
+New-Item -Type Directory -Path $ACCESS_GPO_PATH
+New-Item -Path $ACCESS_GPO_PATH -Name "GptTmpl.inf" -ItemType "file" -Value $ACCESS_SECURITY_TEMPLATE
+New-Item -Path "$env:SystemRoot\SYSVOL\sysvol\$DOMAIN_NAME\Policies\{$ACCESS_POLICY_GUID}\Machine" -Name "comment.cmtx" -ItemType "file" -Value $COMMENT_XML
+
+# Firewall
+$FIREWALL_USER_MODE_IN_TCP = "v2.31|Action=Allow|Active=TRUE|Dir=In|Protocol=6|LPort=3389|App=%SystemRoot%\system32\svchost.exe|Svc=termservice|Name=@FirewallAPI.dll,-28775|Desc=@FirewallAPI.dll,-28756|EmbedCtxt=@FirewallAPI.dll,-28752|"
+Set-GPRegistryValue -Name $ACCESS_GPO_NAME -Key "HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\WindowsFirewall" -ValueName "PolicyVersion" -Type DWORD -Value 543
+Set-GPRegistryValue -Name $ACCESS_GPO_NAME -Type String -Key "HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\WindowsFirewall\FirewallRules" -ValueName "RemoteDesktop-UserMode-In-TCP" -Value $FIREWALL_USER_MODE_IN_TCP
+
+
+# Allow remote RDP connections
+Set-GPRegistryValue -Name $ACCESS_GPO_NAME -Key "HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows NT\Terminal Services" -ValueName "fDenyTSConnections" -Type DWORD -Value 0
+Set-GPRegistryValue -Name $ACCESS_GPO_NAME -Key "HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows NT\Terminal Services" -ValueName "UserAuthentication" -Type DWORD -Value 0
+
+
+# # Step 5/7. Export your LDAP CA certificate
+certutil "-ca.cert" windows.der
+certutil -encode windows.der windows.pem
+
+gpupdate.exe /force
+
+$CA_CERT_PEM = Get-Content -Path windows.pem
+$CA_CERT_PEM_ONE_LINE = $CA_CERT_PEM -join "\n"
+
+
+$NET_BIOS_DOMAIN= (Get-ADDomain).NetBIOSName
+$LDAP_USERNAME = "$NET_BIOS_NAME\$SAM_ACCOUNT_NAME"
+
+# TODO: figure out how to get LDAP address, maybe ask user?
+# this method only for testing
+$PUBLIC_IP=(Invoke-RestMethod http://ipinfo.io/json).ip
+$LDAP_ADDR="$PUBLIC_IP" + ":636"
+
+$DESKTOP_ACCESS_CONFIG_YAML=@'
+windows_desktop_service:
+  enabled: yes
+  listen_addr: "localhost:3028"
+  ldap:
+    addr:     '{0}'
+    domain:   '{1}'
+    username: '{2}'
+    insecure_skip_verify: false
+    pem_ca_cert: "{3}"
```

Comment:
Does powershell have a proper YAML encoder? Instead of listing this all as one long line it would be nice to include the newlines:

```
  pem_ca_cert: |
    -----BEGIN CERTIFICATE-----
    asdfasdfasdf
    ...
    -----END CERTIFICATE-----
```

Reply:
I agree! I tried to make it look nicer, but couldn't find a way to do it in the powershell. Unfortunately powershell does not have YAML encoder / parser.

Reply:
See https://github.com/gravitational/teleport/pull/15842#discussion_r961049171 for what I did to make this look nicer.

Reply:
Nice! I'm glad that you've found a way to properly construct that yaml file. Looks way better.